lib/checks/aria/aria-allowed-attr-evaluate.js

```diff
@@ -27,31 +33,80 @@ import { getRole, allowedAttr, validateAttr } from '../../commons/aria';
  */
 function ariaAllowedAttrEvaluate(node, options, virtualNode) {
   const invalid = [];
-
+  const rowAttrInvalid = [];
   const role = getRole(virtualNode);
   const attrs = virtualNode.attrNames;
   let allowed = allowedAttr(role);
-
   // @deprecated: allowed attr options to pass more attrs.
   // configure the standards spec instead
   if (Array.isArray(options[role])) {
     allowed = uniqueArray(options[role].concat(allowed));
   }
 
+  let tableRole;
+  function validateRowAttrs(validValue) {
+    // check if the parent exists otherwise a TypeError will occur (virtual-nodes specifically)
+    if (virtualNode.parent && role === 'row' && validValue) {
+      // TODO: look into memoizing getRole and removing this local cache
+      let tableMap = cache.get('aria-allowed-attr-table');
+      if (!tableMap) {
+        tableMap = new WeakMap();
+        cache.set('aria-allowed-attr-table', tableMap);
+      }
+
+      const table = closest(
+        virtualNode,
+        'table, [role="treegrid"], [role="table"], [role="grid"]'
+      );
+      tableRole = tableMap.get(table);
+      if (table && !tableRole) {
+        tableRole = getRole(table);
+        tableMap.set(table, tableRole);
+      }
+
+      if (['table', 'grid'].includes(tableRole) && role === 'row') {
+        return true;
+      }
+    }
+  }
+
+  const preChecks = {
```

Comment:
I think this list should be configurable in some way. Maybe as options on the check?

Reply:
I have added a options check with the latest commit `ariaAttrs` which are configurable